src/Action/CompareAction.php

```diff
@@ -46,8 +46,7 @@ public function __invoke(Request $request, string $className, string $id, ?int $
             $newRev = $request->query->get('newRev');
         }
 
-        $ids = explode(',', $id);
-        $diff = $this->auditReader->diff($className, $ids, $oldRev, $newRev);
+        $diff = $this->auditReader->diff($className, $id, $oldRev, $newRev);
```

Comment:
Array are supposed to be supported https://github.com/sonata-project/EntityAuditBundle/blob/e38bb57ea321f59960fd38fe8f88050763524635/src/AuditReader.php#L617
and find support array too https://github.com/sonata-project/EntityAuditBundle/blob/e38bb57ea321f59960fd38fe8f88050763524635/src/AuditReader.php#L220

I would say we try to support something like `3,4,5`.

Do it mean this case is broken https://github.com/sonata-project/EntityAuditBundle/blob/e38bb57ea321f59960fd38fe8f88050763524635/src/AuditReader.php#L235 ?

@wbloszyk @phansys you're using this bundle, do you pass array/multiple ids some times ?

Reply:
If we pass `$ids` as an array to the `find()` function, then inside https://github.com/sonata-project/EntityAuditBundle/blob/e38bb57ea321f59960fd38fe8f88050763524635/src/AuditReader.php#L235-L237 the values are:

* `$idKeys = [0 => 0]`
* `$columnName = 0`

...which leads to https://github.com/sonata-project/EntityAuditBundle/blob/e38bb57ea321f59960fd38fe8f88050763524635/src/AuditReader.php#L246 being

`$whereSQL = "e.rev <= ? AND e.0 = ?"`

While I think the type of `$id` at https://github.com/sonata-project/EntityAuditBundle/blob/e38bb57ea321f59960fd38fe8f88050763524635/src/AuditReader.php#L220 is correct, I think for the `diff()` function at https://github.com/sonata-project/EntityAuditBundle/blob/e38bb57ea321f59960fd38fe8f88050763524635/src/AuditReader.php#L617 it should be `* @param int|string $id` - because I _think_ a diff is on a single object, by definition.

In that case, the new value becomes:

`$whereSQL = "e.rev <= ? AND e.id = ?"` which is correct.

Reply:
...I've added this change into the PR, as `CompareAction` is the only place which calls the `diff()` function in the `AuditReader`.

Reply:
>@wbloszyk @phansys you're using this bundle, do you pass array/multiple ids some times ?

I'm not using composite ids with this bundle.
This issue was reported before. See #288, #331 #389.